servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractRelativeReferences.java

```diff
@@ -123,143 +129,331 @@ public void createReferenceUnambiguous() {
   }
 
   /**
-   * Can create a reference from a hash that is unambiguous and DETACHED. Expect a 200 response with
-   * the created reference.
+   * Cannot create a reference from a branch + hash that is ambiguous (writing operation). The hash
+   * must contain a starting commit ID. Expect a BAD_REQUEST error.
    */
   @Test
   @NessieApiVersions(versions = NessieApiVersion.V2)
-  public void createReferenceUnambiguousDetached() {
+  public void createReferenceFromBranchAmbiguous() {
```

Comment:
Wonder if we need these tests as "pure REST tests" instead of using the Nessie API. I think, it's better to use the Nessie API and only have "pure REST tests" for test cases that do not work via the API.

Reply:
I actually went for REST tests precisely because we can test any test case by forging the HTTP request by hand. Some of the test cases would indeed be untestable through the API.

Reply:
I agree. The java client API seems weaker than the REST API in terms of expressing relative references. It might be worth adding specific methods for "merge parent #N", "ancestor #Y", etc., where hashes are manipulated.

Reply:
> Some of the test cases would indeed be untestable through the API.

fair